pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,134 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/operator-framework/combo/api/v1alpha1"
+	combinationPkg "github.com/operator-framework/combo/pkg/combination"
+	comboConditions "github.com/operator-framework/combo/pkg/conditions"
+	templatePkg "github.com/operator-framework/combo/pkg/template"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
 
-	"github.com/operator-framework/combo/api/v1alpha1"
+var (
+	RequeueDefaultTime = time.Second * 2
+	SuccessCondition   = comboConditions.ProccessedCombinationsCondition
 )
 
 type combinationController struct {
 	client.Client
 	log logr.Logger
 }
 
+// manageWith creates a new instance of this controller
 func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
 	c.log = c.log.V(version)
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Combination{}).
 		Complete(c)
 }
 
-func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request) (reconcile.Result, error) {
-	// Set up a convenient log object so we don't have to type request over and over again
+// Reconcile manages incoming combination CR's and processes them accordingly
+func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request) (result reconcile.Result, deferredErr error) {
+	// Set up a convenient log object so we don’t have to type request over and over again
 	log := c.log.WithValues("request", req)
-	log.Info("reconciling combination")
 
-	in := &v1alpha1.Combination{}
-	if err := c.Get(ctx, req.NamespacedName, in); err != nil {
+	log.Info("new combination event inbound")
+
+	// Attempt to retrieve the requested combination CR
+	combination := &v1alpha1.Combination{}
+	err := c.Get(ctx, req.NamespacedName, combination)
+	if err != nil {
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// If combination is being deleted, remove from queue
+	if !combination.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("combination is being deleted, ignoring event")
+		return reconcile.Result{}, nil
+	}
+
+	log.Info(fmt.Sprintf("combination %s successfully found in reconciler", combination.Name))
+
+	// Attempt to retrieve the template referenced in the combination CR
+	template := &v1alpha1.Template{}
+	templateQuery := types.NamespacedName{Name: combination.Spec.Template}
```

Comment:
nit: I'd personally deflate this templateQuery variable into the c.Get call as we're only using it in a single place (and I don't think it drastically improves readability).